dags/inference/maxtext_inference.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run MaxText inference benchmarks with nightly version."""
+
+import datetime
+from airflow import models
+from dags import composer_env, test_owner
+from dags.vm_resource import TpuVersion, Zone, Project, V5_NETWORKS, V5E_SUBNETWORKS, RuntimeVersion
+from dags.inference.configs import maxtext_inference_gce_config
+from dags.multipod.configs.common import SetupMode, Platform
+
+
+# Run once a day at 4 am UTC (8 pm PST)
+SCHEDULED_TIME = "0 4 * * *" if composer_env.is_prod_env() else None
+
+
+with models.DAG(
+    dag_id="maxtext_inference",
+    schedule=SCHEDULED_TIME,
+    tags=["inference_team", "maxtext", "nightly", "benchmark"],
+    start_date=datetime.datetime(2024, 1, 19),
+    catchup=False,
+) as dag:
+  test_name_prefix = "maxtext-inference"
+  test_models = {
+      "llama2-7b": "llama2_7b",
+      # "mistral": "test_mistral",
+      # "gemma-2b": "gemma/2b/test_gemma",
+      # "gpt3": "test_gpt3",
+  }
+
+  test_mode = SetupMode.NIGHTLY
+
+  # TODO(yeandy) Setup peer network, and then change to CLOUD_TPU_INFERENCE_TEST
```

Comment:
FYI: we have some [notes]([go/ml-auto-solutions-user-guide](http://goto.google.com/ml-auto-solutions-user-guide)) for peer network setup

Reply:
Yup, I found the peer network instructions just after writing this comment 😆 

I think we'll do the peering in a separate PR.